src/Trial.hs

```diff
@@ -52,6 +53,8 @@ data Trial e a
     | Result [e] a
     deriving stock (Show, Eq)
 
+type TaggedTrial tag a = Trial tag (tag, a)
```

Comment:
Interesting approach :thinking: There's another possibility to define these two types:

```haskell
data TaggedTrial tag e a
    = Fiasco [(Fatality, e)]
    | Result [e] tag a

type Trial = TaggedTrial ()
```

With this approach, we can have instance like `Functor` to work automatically with `a`. But this approach also makes the data type more difficult. And we lose a connection between having `e` and `tag` the same. I guess, if we really need, we can define our own functions `fmapTagged`, etc.

Reply:
Oh, having `fmapTagged` is a great idea!
And yes, good point! I think we can have it like this for now and see what the usage priorities will be, so maybe we could adjust the types as needed.